### ISSUE TITLE
Restarting google cloud machine changes its IP and invalidates its certificate

### ISSUE THREAD

**Issue by fikovnik** [docker/machine#675]
I created a machine using:

```
$ docker-machine create -d google --google-project my-project --google-machine-type n1-standard-1 --google-zone europe-west1-b gc-datalyse-1
```

afterwards I can do 
```
$ $(docker-machine env)
$ docker ps
```
and works as expected. However when I restart the machine using `docker-machine restart` it actually recreates the machine, but does not recreate the certificates. If I then do the same command sequence
```
$ $(docker-machine env)
$ docker ps
```
I get
```
FATA[0000] An error occurred trying to connect: Get https://23.251.139.54:2376/v1.17/containers/json: x509: certificate is valid for 104.155.11.62, not 23.251.139.54 
```
The reason is that the IP has changed. I guess it will be easy to regenerate the certificates, but I was wondering why the restart action just does not reset the instance like calling `gcloud compute instances reset <instance>` which will only reboot not?

**Comment by flurdy** [comment]
This is the same issue with Amazon ec2, and I presume any cloud provider where the IP might change if you stop an instance.

**Comment by flurdy** [comment]
Though it is a known issue as in this case with the VMWare Fusion driver: https://github.com/docker/machine/issues/430 and

**Comment by ehazlett** [comment]
This has been fixed in #770
